teps/0090-matrix.md

```diff
@@ -596,17 +627,546 @@ For example:
+    default:
+  tasks:
+  - name: clone
+    taskRef:
+      name: git-clone
+    ...
+  - name: get-test-shards
+    taskRef:
+      name: get-test-shards
+    runAfter:
+    - clone
+    ...
+  - name: test
+    taskRef:
+      name: test
+    matrix:
+    - name: shards
+      value: $(tasks.get-test-shards.results.shards)
+    ...
+  - name: notify
+    taskRef:
+      name: notify
+    params:
+    - name: messages
+      value: $(tasks.test.results.output)
+    ...
+```
+
+After [TEP-0076][tep-0076] is implemented, we will explore this further in
+another TEP. In the meantime, we will disallow consuming `Results` from a
+dynamic matrixed `PipelineTask` given the limitation discussed above.
+
+### Matrix in the Directed Acyclic Graph
+
+When a `PipelineTask` that has a dependency on a matrixed `PipelineTask`, 
+it will not be executed until all the `TaskRuns` or`Runs` generated from 
+the `Matrix` have been executed.
+
+There are two types of dependencies between `PipelineTasks`:
+* Ordering: declared using `runAfter`.
+* Resource: established by passing resources, such as `Results`.
+
+#### Ordering Dependencies
+
+In the example below, the *test* `PipelineTask` is fanned out using the *shards* 
+`Result` in the `Matrix`. The *build* `PipelineTask` is ordering dependent on 
+the *test* `PipelineTask`, based on `runAfter`. As such, all the *test* `TaskRuns` 
+have to complete execution successfully before *build* `PipelineTask` is executed.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline
+spec:
+  tasks:
+  - name: clone
+    taskRef:
+      name: git-clone
+    ...
+  - name: get-test-shards
+    taskRef:
+      name: get-test-shards
+    runAfter:
+    - clone
+    ...
+  - name: test
+    taskRef:
+      name: test
+    matrix:
+    - name: shards
+      value: $(tasks.get-test-shards.results.shards)
+    ...
+  - name: build
+    taskRef:
+      name: build
+    runAfter:
+    - test
+    ...
+```
+
+#### Resource Dependencies
+
+In the example below, the *test* `PipelineTask` is fanned out using the *browsers*
+`Parameter` in the `Matrix`. The *notify* `PipelineTask` is resource dependent on
+the *test* `PipelineTask`, based on *output* `Results`. As such, all the *test* 
+`TaskRuns` have to complete execution successfully before *notify* `PipelineTask` 
+is executed.
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: pipeline
+spec:
+  params:
+  - name: browsers
+    type: array
+    default:
+      - chrome
+      - firefox
+      - safari
+  tasks:
+  - name: clone
+    taskRef:
+      name: git-clone
+    ...
+  - name: test
+    taskRef:
+      name: test
+    matrix:
+    - name: browsers
+      value: $(params.browsers)
+    ...
+  - name: notify
+    taskRef:
+      name: notify
+    params:
+    - name: messages
+      value: 
+      - $(tasks.test-1.results.output)
+      - $(tasks.test-2.results.output)
+      - $(tasks.test-3.results.output)
```

Comment:
I understood we wouldn't support results in the first implementation?

Reply:
that's right, won't be supported in the first implementation but we discuss some possibilities in the proposal